runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -52,11 +63,46 @@
 	@Inject
 	private ClanManager clanManager;
 
-	@Schedule(
-		period = 600,
-		unit = ChronoUnit.MILLIS
-	)
-	public void updateClanChatTitle()
+	@Inject
+	private ClanChatConfig config;
+
+	private List<String> chats = new ArrayList<>();
+
+	private static final int MAX_CHATS = 10;
+	private static final String RECENT_TITLE = "Recent Clan Chats";
+
+	@Provides
+	ClanChatConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(ClanChatConfig.class);
+	}
+
+	@Override
+	public void startUp()
+	{
+		if (!config.chatsData().isEmpty())
+		{
+			chats = Arrays.stream(config.chatsData().split(",")).collect(Collectors.toList());
```

Comment:
Splitter.on(",").splitToList() ?

Reply:
Splitter.On() creates an immutable list.

Reply:
I see you remedied that.